src/CollabVM.cpp

```diff
@@ -1848,6 +1850,17 @@ void CollabVMServer::OnConnectInstruction(const std::shared_ptr<CollabVMUser>& u
 
 	VMController& controller = *it->second;
 
+	if(!controller.GetSettings().VMPassword.empty())
+	{
+		// Check if the password is valid (or if the password is there at all)
+		if(args.size() != 2 || args[1] != controller.GetSettings().VMPassword) {
+			// Invalid password
+			SendWSMessage(*user, "7.connect,1.3;"); // 2 is used by something(?) in the webapp code, so let's go with 3 instead
+			return;
+		}
+		// Password is correct, continue.
```

Comment:
This comment is probably unneeded, as the general control flow is obvious based on other comments within vicinity.

Reply:
The comment about not knowing what 7.connect,1.2 leads to is a bit of a dumb action, considering discreet comments, both server-side and client-side, explain that connect opcode 2, is disconnect.

C: (/src/js/collab-vm/collab-vm.js)```js
guac.onconnect = function(parameters) {
...
case 2: // Disconnected```

S: (/src/CollabVM.cpp) ```cpp
if (args.empty())
	{
		// Disconnect
		...
		SendWSMessage(*user, "7.connect,1.2;");```